data_model/src/lib.rs

```diff
@@ -650,27 +654,27 @@ pub mod model {
     pub enum RegistrableBox {
         /// [`Peer`](`peer::Peer`) variant.
         #[display(fmt = "Peer {_0}")]
-        Peer(Box<<peer::Peer as Registered>::With>),
+        Peer(<peer::Peer as Registered>::With),
```

Comment:
This isn't exactly equivalent to `Box`-ing on the outside. Consider that the enum before had a fixed size of $2 \times \text{usize}$ and now has the size of the biggest `Registered::With`. 

As such `Box`-ing `val.into()` defeats the whole point of boxing to reduce wasted RAM space. Consider removing `Box` es entirely.